jsonparser.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <sstream>      // std::istringstream
+#include <map> 
+#include <fstream>
+
+class jsonparser{
+private:
+public:
+    static std::string rFVbQ (std::string st){   // alias: returnFirstValuebetweenQuotationmarks, finds the first substd::string that sits between " marks and returns it
+        if (st.find('"') == std::string::npos) throw "invalid json syntax"; // if doesn't contain throw
+        std::string tmp = st.substr(st.find('"')+1);
+        if (tmp.find('"') == std::string::npos) throw "invalid json syntax"; // if doesn't contain throw
+        return tmp.substr(0,tmp.find('"'));
+    }
+    static std::map<std::string, std::string>  istrmInp(std::stringstream& f){ // read the given istream file and convert the data to a map
+        std::map<std::string, std::string> parsed;
+        std::string t = "";
+        while (true) {
+            getline(f, t);
+            if (t.find('{') != std::string::npos) continue;
```

Comment:
Ez veszélyesnek tűnik, ha egy kulcs még ugyanabban a sorban elkezdődik, amelyikben a { is van. A json nem sok megkötést tartalmaz a fehér karakterek kezelésére.